plugins/PagePerformance/PagePerformance.php

```diff
@@ -0,0 +1,145 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+
+namespace Piwik\Plugins\PagePerformance;
+
+use Piwik\DataTable;
+use Piwik\Plugin\ViewDataTable;
+use Piwik\Plugins\CoreVisualizations\Visualizations\HtmlTable;
+
+/**
+ */
+class PagePerformance extends \Piwik\Plugin
+{
+    public static $availableForMethods = [
+        'getPageUrls',
+        'getEntryPageUrls',
+        'getExitPageUrls',
+        'getPageUrlsFollowingSiteSearch',
+        'getPageTitles',
```

Comment:
So far the data for page titles doesn't seem to be archived? But as mentioned in another comment once we hook into actions archiving this should work out of the box